File: python/src/buble_ai/types/media.py
from __future__ import annotations

from typing import Any, List, Optional, TypedDict


class MediaModelParameter(TypedDict, total=False):
    name: str
    type: str
    label: str
    default: Any
    values: List[Any]
    min: float
    max: float
    step: float
    required: bool


class MediaModelOperation(TypedDict, total=False):
    mode: str
    input: dict
    parameters: List[MediaModelParameter]


class MediaModel(TypedDict, total=False):
    model: str
    name: str
    media_type: str
    operations: List[MediaModelOperation]


class MediaModelsEnvelope(TypedDict):
    data: List[MediaModel]

File: npm/src/types/media.ts
import type { ApiEnvelope, MediaType } from './common.js';

export type MediaModelParameter = {
  name: string;
  type?: string;
  label?: string;
  default?: unknown;
  values?: unknown[];
  min?: number;
  max?: number;
  step?: number;
  required?: boolean;
};

export type MediaModelOperation = {
  mode: string;
  input?: Record<string, unknown>;
  parameters?: MediaModelParameter[];
};

export type MediaModel = {
  model: string;
  name?: string;
  media_type: MediaType | string;
  operations: MediaModelOperation[];
};

export type ListMediaModelsOptions = {
  media_type?: MediaType;
  signal?: AbortSignal;
  timeout?: number;
};

export type MediaModelsEnvelope = ApiEnvelope<MediaModel[]>;
